dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceLoader.java

```diff
@@ -0,0 +1,158 @@
+                    }
+                    case CONTAINER: {
+                        return new ContainerLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case TEMPLATE: {
+
+                        return new TemplateLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case CONTENT: {
+                        return new ContentletLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case FIELD: {
+                        return new FieldLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case CONTENT_TYPE: {
+                        return new ContentTypeLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case SITE: {
+                        return new SiteLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case HTMLPAGE: {
+                        return new PageLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case VELOCITY_MACROS: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case VTL: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case VELOCITY_LEGACY_VL: {
+                        return VTLLoader.instance()
+                            .writeObject(id1, id2, mode, language, filePath);
+                    }
+                    default: {
+                        CacheLocator.getVeloctyResourceCache()
+                            .addMiss(path);
+                        throw new ResourceNotFoundException("Cannot find velocity file : " + path);
+                    }
+                }
+
+            } catch (Exception e) {
+                throw new ResourceNotFoundException("Cannot parse velocity file : " + path, e);
+            }
+        }
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.velocity.runtime.resource.loader.FileResourceLoader#getLastModified(org.apache.
+     * velocity.runtime.resource.Resource)
+     */
+    @Override
+    public long getLastModified(Resource resource) {
+        return 0;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.velocity.runtime.resource.loader.FileResourceLoader#isSourceModified(org.apache.
+     * velocity.runtime.resource.Resource)
+     */
+    @Override
+    public boolean isSourceModified(Resource resource) {
+        return false;
+    }
+
+    public static DotResourceLoader getInstance() {
+        if (instance == null) {
+            synchronized (DotResourceLoader.class) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this dangerous instance of double-checked locking. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2168)